### ISSUE TITLE
[BUG] Max length of volume name not consist between frontend and backend

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#3917]
## Describe the bug

From https://github.com/longhorn/longhorn/issues/3915#issuecomment-1117000306, Maximum volume name length is 40 in the backend, but in UI the max volume name length is 63

## To Reproduce

1. Deploy Longhorn
2. Create volume from UI, set volume name more than 63 characters
3. Click `OK` button
```
Volume.longhorn.io "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-bb1d3fa2" is invalid: metadata.labels: Invalid value: "<64 character volume name>": must be no more than 63 characters
```
## Expected behavior

We should tell user our design no matter it's 63 char or 40 char

## Log or Support bundle

N/A

## Environment

 - Longhorn version: master-head 20220504
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: K3s
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3
 - Node config
   - OS type and version: Ubuntu 20.04
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWARE
 - Number of Longhorn volumes in the cluster: 1

## Additional context

N/A

**Comment by innobead** [comment]
The backend has no strict limitation for the length but truncates the value to the max length of 40 instead only.

This should not be an issue after https://github.com/longhorn/longhorn/issues/3915 is fixed.

**Comment by innobead** [comment]
Even the length correction of the name is expected, but it's implicit actually, so it's better to have this behavior explicitly validated. Just reopen this for the improvement in the future.

**Comment by innobead** [comment]
This should be able to be improved in our validating admission webhook, and it will run before the general metadata label validation.

**Comment by weizhe0422** [comment]
An error occurred while labeling the volume.

After setting the label, the volume name length will be truncated to length 40.

**Comment by innobead** [comment]
The status of the pipeline was not right, so I just moved this to `review`. After updating the ready-for-testing checklist, move this to ready for testing stage.

**Comment by weizhe0422** [comment]
Since there is no mutating mechanism was implemented in v1.2.x, so I check the way to fix this issue.

**Comment by innobead** [comment]
> Since there is no mutating mechanism was implemented in v1.2.x, so I can't move the labeling to mutating like mater and v1.3.x do. I will check the way to fix this issue.

Then, no need to backport to 1.2.x.

**Comment by roger-ryao** [comment]
**Verified on master-head 20221116**

- longhorn master-head ([7b27a9a](https://github.com/longhorn/longhorn/commit/7b27a9ad49e719bf992401c506b5abefa798d8f7)) 
- longhorn-manager master-head ([732f3c8](https://github.com/longhorn/longhorn-manager/commit/732f3c8392ca2f8a394ba001015153a649a31691)) 


**The test steps**

[Ref #3719 (To Reproduce)](https://github.com/longhorn/longhorn/issues/3917#issuecomment-1313589295)

**Result Passed**
- [x] the operations (create volume|attach|create PV&PVC) work normally w/o errors
- [x] the metadata/label of `volume|replica|engine` have longhornvolume: <volume-name>
- [x] the metadata/finalizers of `volume|replica|engine` have longhorn.io
- [x] the volume can be deleted normally
![Screenshot_20221116_141703](https://user-images.githubusercontent.com/104337648/202107200-50fbf24f-9bbe-48f6-8263-808fe3817dc6.png)
![Screenshot_20221116_144201](https://user-images.githubusercontent.com/104337648/202107206-2b1e57c0-40f4-4c06-b567-174854ea0518.png)
![Screenshot_20221116_144722](https://user-images.githubusercontent.com/104337648/202107210-04712fef-8ba0-47e6-b994-ac1dc3cbb5a4.png)
